### PR TITLE
Story 328: Mixed recruiters

### PR DESCRIPTION
**Pull request by davisagli**
These are a few changes to make Griduniverse work with https://github.com/Dallinger/Dallinger/pull/978

### CODE REVIEW

File: test/conftest.py [hunk 1]
```diff
@@ -196,7 +197,13 @@ def participants(db_session):
     from dallinger.models import Participant
     ps = []
     for i in range(10):
-        p = Participant(worker_id=str(i), hit_id='1', assignment_id='1', mode="test")
+        p = Participant(
```
Comment (jessesnyder) [inline test/conftest.py]: Could delegate to the other fixture(?)
```
for i in range(10):
    ps.append(a.participant(worker_id=str(i)))
...
```

File: dlgr/griduniverse/config.txt [hunk 2]
```diff
@@ -26,7 +26,7 @@ database_size = standard-0
 [Server Parameters]
 dyno_type = standard-2x
 num_dynos_web = 1
-num_dynos_worker = 3
+num_dynos_worker = 1
```
[the remark this answers is not in the document]
Comment (jessesnyder) [reply dlgr/griduniverse/config.txt]: Remembered you'd added some notes about this to running_bots.rst in Dallinger... probably the right place for it.

**Comment by jessesnyder** [comment]
When I tried running this with a combination of `bots` and `mturk`, I see the HIT, but only the Heroku "There's nothing here, yet." page. It looks like the bots did finish and perhaps submitted the assignment correctly, because I see this traceback in `dallinger logs`:
```
May 17 12:37:13 dlgr-48578af9 app/web.1:  Traceback (most recent call last): 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app 
May 17 12:37:13 dlgr-48578af9 app/web.1:      response = self.full_dispatch_request() 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request 
May 17 12:37:13 dlgr-48578af9 app/web.1:      rv = self.handle_user_exception(e) 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception 
May 17 12:37:13 dlgr-48578af9 app/web.1:      reraise(exc_type, exc_value, tb) 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/python/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise 
May 17 12:37:13 dlgr-48578af9 app/web.1:      raise value 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request 
May 17 12:37:13 dlgr-48578af9 app/web.1:      rv = self.dispatch_request() 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request 
May 17 12:37:13 dlgr-48578af9 app/web.1:      return self.view_functions[rule.endpoint](**req.view_args) 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/src/dallinger/dallinger/db.py", line 88, in wrapper 
May 17 12:37:13 dlgr-48578af9 app/web.1:      return func(*args, **kwargs) 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/src/dallinger/dallinger/experiment_server/experiment_server.py", line 1600, in worker_complete 
May 17 12:37:13 dlgr-48578af9 app/web.1:      _worker_complete(participant_id) 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/src/dallinger/dallinger/experiment_server/experiment_server.py", line 1625, in _worker_complete 
May 17 12:37:13 dlgr-48578af9 app/web.1:      event_type=event_type, 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/src/dallinger/dallinger/experiment_server/experiment_server.py", line 1572, in _handle_worker_event 
May 17 12:37:13 dlgr-48578af9 app/web.1:      return worker_function(event_type, assignment_id, participant_id) 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/src/dallinger/dallinger/db.py", line 88, in wrapper 
May 17 12:37:13 dlgr-48578af9 app/web.1:      return func(*args, **kwargs) 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/src/dallinger/dallinger/experiment_server/experiment_server.py", line 1764, in worker_function 
May 17 12:37:13 dlgr-48578af9 app/web.1:      runner() 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/src/dallinger/dallinger/experiment_server/worker_events.py", line 75, in __call__ 
May 17 12:37:13 dlgr-48578af9 app/web.1:      self.approve_assignment() 
May 17 12:37:13 dlgr-48578af9 app/web.1:    File "/app/.heroku/src/dallinger/dallinger/experiment_server/worker_events.py", line 107, in approve_assignment 
May 17 12:37:13 dlgr-48578af9 app/web.1:      self.participant.recruiter.approve_hit(self.assignment_id) 
May 17 12:37:13 dlgr-48578af9 app/web.1:  AttributeError: 'MultiRecruiter' object has no attribute 'approve_hit' 
```

**Comment by davisagli** [comment]
Yeah, I also saw the issue with bots not submitting the questionnaire, which I don't think is related to these changes.

That traceback on heroku indicates that a valid recruiter wasn't passed in the request when the participant was created, so it fell back to using the recruiter from config. Maybe your browser was using an older cached version of the griduniverse js without these fixes?

**Pull request merged**